ci_group/revolve2/ci_group/morphological_novelty_metric/_build_cmodule.py

```diff
@@ -0,0 +1,77 @@
+import os
+import shutil
+from glob import glob
+from os.path import join
+
+import numpy
+from Cython.Build import cythonize
+from setuptools import Extension, setup
+
+
+def build() -> None:
+    """
+    Build the morphological novelty shared object.
+
+    :raises OSError: If the users OS is not Windows or UNIX-based.
+    """
+    directory_path = os.path.dirname(os.path.abspath(__file__))
+
+    source = join(directory_path, "_calculate_novelty.pyx")
+    include = numpy.get_include()
+
+    match os.name:
+        case "nt":  # Windows
+            extra_compile_args = [
+                "/O2",
+                "-UNDEBUG",
+            ]
+        case "posix":  # UNIX-based systems
+            extra_compile_args = [
+                "-O3",
+                "-Werror",
+                "-Wno-unreachable-code-fallthrough",
+                "-Wno-deprecated-declarations",
+                "-Wno-parentheses-equality",
+                "-ffast-math",
+                "-UNDEBUG",
+            ]
+        case _:
+            raise OSError(
+                f"No build parameter set for operating systems of type {os.name}"
+            )
+
+    ext = Extension(
+        name="calculate_novelty",
+        sources=[source],
+        include_dirs=[include],
+        define_macros=[("NPY_NO_DEPRECATED_API", "NPY_1_7_API_VERSION")],
+        extra_compile_args=extra_compile_args,
+    )
+
+    ext_modules = cythonize(
+        ext,
+        include_path=[include],
+        compiler_directives={"binding": True, "language_level": 3},
+    )
+
+    setup(
+        ext_modules=ext_modules,
+        script_args=["build_ext", "--inplace"],
+    )
+
+    curr_path = os.getcwd()
+    outputs = glob(f"{curr_path}/*.so")
```

Comment:
Can either check for .so and .dll maybe. but alternatively you can look at something like
```
class CustomBuildExt(build_ext):
    def get_ext_fullpath(self, ext_name):
        # Set the output directory to be the same as the directory of the .pyx file.
        output_dir = os.path.dirname(os.path.abspath(__file__))
        filename = self.get_ext_filename(ext_name)
        return os.path.join(output_dir, filename)

    def run(self):
        # Run the original build_ext commands/
        super().run()

        # Set execute permissions on the created files.
        for ext in self.extensions:
            fullname = self.get_ext_fullname(ext.name)
            filename = self.get_ext_fullpath(fullname)
            if os.path.exists(
                filename
            ):  # File cannot exist because of e.g. build failure.
                current_mode = os.stat(filename).st_mode
                os.chmod(filename, current_mode | 0o0555)

(...)
setup(
    ext_modules=ext_modules,
    cmdclass={'build_ext': CustomBuildExt},
)
(...)
if __name__ == "__main__":
    if "build_ext" not in sys.argv:
        sys.argv.append("build_ext")
    build()
```

definitely needs testing, but this way you get the artifact paths directly

Reply:
now i removed this block alltogether, and just clarified the build directory for the script